plcx/comm/server.py

```diff
@@ -1,14 +1,15 @@
 import asyncio
 import logging
-from typing import Callable
+from typing import Callable, Coroutine, Union
 
 from plcx.constants import MAX_TRY
 from plcx.exceptions import NotReadableMessage
+from plcx.utils.coroutine import await_if_coroutine
 
 logger = logging.getLogger(__name__)
 
 
-def tcp_read_echo(response_handler: Callable, read_bytes: int = 512) -> asyncio.coroutine:
+def tcp_read_echo(response_handler: Union[Callable, Coroutine], read_bytes: int = 512) -> asyncio.coroutine:
```

Comment:
`response_handler` is not a `Coroutine`. It is a "coroutine function", meaning that whatever it returns is a coroutine. Coroutine is something, that you can await.

I'm not sure there exists a pre-defined type for coroutine functions. if we want to type this, we need to type it as a `Callable` which can return either `None` or a `Coroutine`.